spynnaker/pyNN/models/neuron/plasticity/stdp/common.py

```diff
@@ -51,3 +51,41 @@ def get_exp_lut_array(time_step, time_constant, shift=0):
     # Concatenate with the header
     header = numpy.array([len(a), shift], dtype="uint16")
     return numpy.concatenate((header, a.astype("uint16"))).view("uint32")
+
+
+def _get_last_non_zero_value(values):
+    """ Get the last non-zero value (rescaled) from a LUT array
+    """
+    # Either the ultimate or penultimate value must be non-zero as generated
+    # from the above function
+    if values[-1] != 0:
+        return values[-1] / STDP_FIXED_POINT_ONE
+    return values[-2] / STDP_FIXED_POINT_ONE
+
+
+def get_min_lut_value(
+        exp_lut_array, time_step=None, max_stdp_spike_delta=None):
+    """ Get the smallest non-zero value of an exponential lookup array,\
+        or None if no such value
+
+    :param numpy.ndarray exp_lut_array: The lookup array
+    :param float time_step: The time step in milliseconds
+    :param float max_stdp_spike_delta: The maximum expected difference between
+        spike times in milliseconds
+    :rtype: float
+    """
+    if not len(exp_lut_array):
```

Comment:
Just do:
if not exp_lut_array:

Reply:
I'm not sure this is possible - but also not sure that this check is necessary.  This function is only ever called if exp_lut_array exists, so I'm not sure what this was checking.

Reply:
I don't think it can be empty provided that STDP_FIXED_POINT_ONE is a sensible value... but I suppose it could be possible that it's empty.  But we don't have any examples where it is empty, and I suspect if it does return None then it'll lead to a crash elsewhere because it'll be trying to multiply by the returned None value.

Reply:
If None fails elsewhere, better to fail early and pick up the real error then

Reply:
Agreed. if not exp_lut_array:  fails   
Had not seen it was a numpy array! Sorry

A numpy purest would say do:
if not exp_lut_array.size
but len(exp_lut_array) works too

I also 100% agree don't return None if calling code can not handle None!